streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java

```diff
@@ -61,7 +62,7 @@
      * @throws IllegalStateException If store gets registered after initialized is already finished
      * @throws StreamsException if the store's change log does not contain the partition
      */
-    void init(ProcessorContext context, StateStore root);
+    void init(ProcessorContext context, StateStore root, StreamsConfig config);
```

Comment:
`ProcessorContext#appConfigs()` gives you access to `StreamsConfig#originals()`. Thus I think that we can avoid to change the API (note, this would be a public API change and require a KIP).